### DISCUSSION TITLE
Serial data Rate and Sensors

### DISCUSSION THREAD

**Discussion by Wolfillo** [metrafonic/asyncio-vedirect-mqtt#4] [Q&A] [answered]
Hi again! 

I'm still learning from how this system works and also deep-diving in the RasPi world. Beeing solved the system service problem I had, I have some questions regarding the serial ports and the sensors provided by the script.

For the first part, I have a RasPi Zero W, which, by default, uses the miniUART (ttyS0) for serial coms. As far as I know this would't be the optimal setup, so I switched off the bluetooth and the raspi automatically assigned ttyAMA0 to the main UART. I also configured the UART 19200 baud rate, though I saw no difference in data rate. My question here is, how often the BlueSolar sends it's data? Is it baudrate dependant? Did I waste my time changing the UART?

On the other hand, searching documentation for the ve.direct protocol, I saw that the MPPT sends more data than this software provides. For example:
PID	0xA043			-- Product ID for BlueSolar MPPT 100/15
FW	119			-- Firmware version of controller, v1.19
SER#	HQXXXXXXXXX		-- Serial number
V	13790			-- Battery voltage, mV
I	-10			-- Battery current, mA
VPV	15950			-- Panel voltage, mV
PPV	0			-- Panel power, W
CS	5			-- Charge state, 0 to 9
ERR	0			-- Error code, 0 to 119
LOAD	ON			-- Load output state, ON/OFF
IL	0			-- Load current, mA
H19	0 			-- Yield total, kWh
H20	0			-- Yield today, kWh
H21	397			-- Maximum power today, W
H22	0 			-- Yield yesterday, kWh
H23	0			-- Maximum power yesterday, W
HSDS	0			-- Day sequence number, 0 to 365
Checksum	l:A0002000148		-- Message checksum

Also my BlueSolar appears as SmartSolar with Firmware 1.0, which is not the case.

Thanks again for your effort.

**Comment by metrafonic** [comment, marked as answer]
Hi! Sorry for the late reply!
Regarding the baud rate, this is not something that can be negotiated with the victron device, so we must "tune in" on the same baud rate that the device is sending, which is 19200. With regard to time, the device sends data around every two seconds using the ve.direct text mode. HOWEVER, in this project, we send a moving average value every minute to avoid sending data every couple of seconds, just to save bandwidth. You can open up an issue for varying the update interval if you like.

And yes you are correct that the device sends more data. I have only selected a handful of stats that will be parsed, as some stats are only available on specific models. Also, the model is hard coded and not parsed as of now. 

Ill look into doing a refactor of the code soon, and include some of these good issues you identified.

**Comment by metrafonic** [comment]
I added the `--windowing` flag to the program, which can be tested on the development branch using:
```
pip3 install git+https://github.com/metrafonic/asyncio-vedirect-mqtt.git@development
```
`ve-direct --windowing 1`  should get you quite instant results. I havent tested this yet

**Reply by metrafonic** [reply]
Its now part of the master branch, so no need for the @development part in the url

**Reply by Wolfillo** [reply]
Hi! Perfect, I'll try when possible and will comment the results!

Thank you.
